src/java/com/rapleaf/jack/AbstractDatabaseModel.java

```diff
@@ -608,43 +607,124 @@ public void clearForeignKeyCache() {
   protected abstract void setAttrs(T model, PreparedStatement stmt)
       throws SQLException;
 
-  @Override
-  public boolean save(T model) throws IOException {
+  // Placeholder default ActiveRecord lock field name
+  private String getLockFieldName(T model) {
+    return "lock_version";
+  }
+
+  // ActiveRecord documentation says it supports optimistic locking using integer fields
+  // That corresponds to the Integer and Long fields, but we only support locking on Integer for now
+  private boolean supportsOptimisticLocking(T model) {
+    final String lockFieldName = getLockFieldName(model);
+
+    if (model.hasField(lockFieldName)) {
+      Object field = model.getField(lockFieldName);
+      return (field != null && field.getClass().equals(Integer.class));
+    } else {
+      return false;
+    }
+  }
```

Comment:
Will `getLockFieldMethod` be used somewhere else in the future? Looks like `getLockFieldName` could be replaced with a static constant class variable (maybe in `JackUtility`), and `supportsOptimisticLocking` could be simplified as:
```java
return model.hasField(LOCK_FIELD_NAME) && Integer.class.equals(model.getField(LOCK_FIELD_NAME));
```